py-polars/polars/dataframe/plotting.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import importlib
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from polars import DataFrame
+
+
+def _get_backend(name: str | None) -> Any:
+    if name is None:
+        raise TypeError(
+            "Please specify a plotting backend using the argument `backend`"
+        )
+    module = importlib.import_module(name)
+    if hasattr(module, "plot"):
+        return module
+    raise ValueError(f"Backend {name} does not have a plot function")
+
+
+class PlotNameSpace:
+    """
+    DataFrame.plot namespace.
+
+    Polars does not implement plotting logic itself. Rather, the idea is:
+
+    - if you use any plotting functions or arguments as documented here, then you
+      should expect consistent behaviour supported backends. Any extra keyword arguments
+      you specify are passed directly to the backend itself, and so their behaviour is
+      backend-dependent.
+    - Polars plots are only intended as convenience methods for ease of use during
+      interactive development.
+
+    In order for a plotting backend to be supported, it must:
+    - implement a top-level `plot` method
+    - the signature of `plot` must be compatible with (but not limited to):
+
+      .. code-block:: python
+
+          def plot(data: DataFrame, kind: str, **kwargs: Any):
+              ...
+    - plotting backends are developed and maintained outside of
+      Polars. Backends currently known to work are:
+
+      - plotly
+      - hvplot
+
+      but once we set out the rules, others may be on their way.
+
+    Polars plotting methods are only intended as convenience methods for ease of
+    use during interactive development. For more complex plots, it is recommended
+    to use plotting libraries directly.
```

Comment:
I like this idea. Can we make explicit here that we don't guarantee anything here. It is meant for EDA and we don't want a major version bump if we change anything here.